pkg/handlers/host.go

```diff
@@ -107,16 +109,33 @@ func (h *HostHandlers) PatchHostByID(w http.ResponseWriter, req *http.Request) e
 	}
 	hostToDB, err := h.constructHostForDB(createHostRequest, req)
 	if err != nil {
-		return err
+		slog.Error("Failed to constructHostForDB", slog.Any("error", err))
+		return api.WriteJSON(w, http.StatusInternalServerError, api.APIError{Error: err.Error()})
 	}
 	hostToDB.ID = id
+	hostGet, err := h.hostService.GetHostByID(hostToDB.ID)
+	if err != nil {
+		statusCode := http.StatusNotFound
+		return api.WriteJSON(w, statusCode, api.APIError{Error: http.StatusText(statusCode)})
+	}
+
+	// Get the value of the host, IP if it's an IP type, Hostname if it's a Domain/Subdomain
+	if errValidation := h.hostService.ValidateHost(createHostRequest.Value); errValidation != nil {
+		// Handle the case when the host value (the target) is not valid
+		slog.Error("Failed to validate host value", slog.String("host_value", createHostRequest.Value))
```

Comment:
Logging the error here would come in handy too 😉

 ```suggestion
		slog.Error("Failed to validate host value", slog.String("host_value", createHostRequest.Value), slog.Any("error", errValidation))
```